servicetalk-concurrent-api/docs/modules/ROOT/pages/blocking-implementation.adoc

```diff
@@ -20,77 +21,186 @@ Everything inside ServiceTalk is somehow connected to one of the three asynchron
+<5> The application thread will continue to publish all the elements from the source and will finally send the
+`onComplete()` signal. This signal will also be offloaded to the `publisherExecutor` and will complete the `Future` with
+the integer collection result.
 
-- Wrap the `Subscriber` such that all `Subscriber` and `Subscription` method calls are offloaded to a chosen
-`Executor`. This mode is for operators that process signals from the original `Publisher` asynchronously.
-- Do not wrap the `Subscriber` and forward all `Subscriber` and `Subscription` methods directly on the calling thread to
-avoid the overhead of wrapping each and every `Subscriber` and `Subscription` in a chain. This mode is for operators
-that process signals from the original `Publisher` synchronously.
+<6> The application thread will wait at `get()` for the `Future` result to be asynchronously completed.
 
-Taking the same example from xref:{page-version}@servicetalk-concurrent-api::blocking-safe-by-default.adoc[here]
+=== `subscribeOn()` Example
+
+Using the `subscribeOn(executor)` operator allows processing of the subscription and demand on a specific thread. Using
+the `subscribeOn(executor)` operator generally requires an understanding of the behavior of the source; using a
+different source may change the need for offloading. `subscribeOn(executor)` is used less frequently than
+`publishOn(executor)` but is useful when it is necessary to offload the control path; the `subscribe` method or
+`Subscription` methods.
 
 [source, java]
 ----
- client.request() # <1>
-       .map(resp -> {
-            return resp.toString(); # <2>
-       })
-       .flatMap(stringResp -> { # <3>
-            return client2.request(stringResp);
-       })
-       .filter(stringResp -> {
-            stringResp.equals("Hello World");  # <4>
-       });
+Collection<Integer> result = Publisher.range(1, 10) <2> <4>
+        .map(element -> element)  // NO-OP
+        .subscribeOn(subscribeExecutor)
+        .toFuture() <3>
+        .get(); <1>
 ----
-<1> A hypothetical client which provides a `request()` method that returns a `Single<Response>`.
-<2> Converting the response to a `String`.
-<3> Call another `client2` that provides a new `Single` which is returned from `flatMap`.
-<4> Only allow "Hello World" messages to be emitted.
 
-In the above example the operators `map` and `filter` will not wrap `Subscriber` and `Subscription` since they do not do
-any asynchronous work. However, `flatmap` will wrap `Subscriber` and `Subscription` to offload the calls to the chosen
-`Executor`.
+<1> `toFuture().get()` will do a `subscribe(Subscriber)`. This flows up the operator chain toward the source;
+`subscribeOn` (offload onto `subscribeExecutor` thread) -> `map` -> `Range`.
+
+<2> `Range` will call `Subscriber.onSubscribe(Subscription)` on the `Subscriber`. This flows back down the operator
```

Comment:
this will also be on the `subscribeExecutor` thread. in <4> this is called out as "Still on a thread from `subscribeExecutor`" ... we should be consistent here.